### PR TITLE
use local version of all case conversion methods

### PR DESCRIPTION
**Pull request by erdoganoksuz**
Signed-off-by: erdoganoksuz <erdoganoksuz95@gmail.com>

## Hey, I just made a Pull Request!

This line converts 'API' to 'api' but if user system language Turkish this converts like this 'API' to 'apı' for this reason 'api' statement not work as expected.

More detail: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLocaleLowerCase

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by erdoganoksuz** [comment]
@OrkoHunter  @freben  i fix prettier issue can you check again?

File: .changeset/pink-knives-obey.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+'example-app': patch
```
Comment (OrkoHunter) [inline .changeset/pink-knives-obey.md:2]: Nit: We don't publish `example-app` package so we can remove this line.

**Review by OrkoHunter — APPROVED** [review]
Looks 👌 

I think we should document this as an [ADR](https://backstage.io/docs/architecture-decisions/adrs-overview) to always use `toLocale(Lower|Upper)Case` over `to(Lower|Upper)Case`.

Similar to [avoiding React.FC](https://backstage.io/docs/architecture-decisions/adrs-adr006).

**Comment by freben** [comment]
@erdoganoksuz Thanks! As the other review mentioned, would be great if you could just remove the example-app line at least. We can make the ADR for it after

**Comment by erdoganoksuz** [comment]
> @erdoganoksuz Thanks! As the other review mentioned, would be great if you could just remove the example-app line at least. We can make the ADR for it after

it would be good for me this week I'm so busy, on the weekend I will create ADR for this. I removed example-app patch can you review it again?

**Review by freben — APPROVED** [review]

**Comment by timbonicus** [comment]
Did an ADR ever come out of this? I looked for it recently, but didn't find one.

**Comment by freben** [comment]
Nope, not yet!

**Comment by iamEAP** [comment]
A lint rule would be a sweet alternative / addition.

**Comment by Rugvip** [comment]
Lint rule sounds good x) #7419

**Pull request merged**